src/journal/models.py

```diff
@@ -574,29 +585,32 @@ def is_serial(self):
     def display_title(self):
         if self.issue_type.code != 'issue':
             return self.issue_title
+        else:
+            return self.pretty_issue_identifier
 
+    def issue_title_parts(self, article=None):
         journal = self.journal
-        issue_identifier = self.pretty_issue_identifier
-
-        title = "{}".format(
-            self.issue_title) if journal.display_issue_title else ""
-
-        title_list = [issue_identifier, title]
-
-        return mark_safe(" &bull; ".join((filter(None, title_list))))
-
-    def issue_title_parts(self):
-        journal = self.journal
-        volume = issue = year = ''
+        volume = issue = year = issue_title = article_number = page_numbers = ''
 
         if journal.display_issue_volume and self.volume:
             volume = ugettext("Volume") + " {}".format(self.volume)
         if journal.display_issue_number and self.issue and self.issue != "0":
             issue = ugettext("Issue") + " {}".format(self.issue)
         if journal.display_issue_year and self.date:
             year = "{}".format(self.date.year)
-
-        return [volume, issue, year]
+        if journal.display_issue_title:
+            issue_title = self.issue_title
+        if journal.display_article_number and article and article.article_number:
+            article_number = ugettext("Article") + " {}".format(article.article_number)
+        if journal.display_article_page_numbers and article:
+            if article.page_range:
+                page_numbers = article.page_range
+            elif article.total_pages:
+                if article.total_pages > 1:
+                    page_numbers = str(self.total_pages) + ugettext('pages')
+                else:
+                    page_numbers = str(self.total_pages) + ugettext('page')
```

Comment:
I believe we need to add a space in these string concatenations, perhaps a format string would make it more clear, though I see it wasn't done that way above.

Reply:
I reworked this and added a few assertion statements to the test to cover the different possible combos.